File: Task_12/ClockPy/Clock.py
print("Helloworlsd")
import tracemalloc
import time


class Counter:
    def __init__(self, _name, _count = 0):
        self.name = _name
        self.count = _count

    def Increment(self):
        self.count += 1

    def Reset(self):
        self.count = 0

    def Ticks(self):
        return self.count

    
      
Count = Counter("Hour Counter")

print(Count.name)


class Clock:
    def __init__(self, hr = Counter("Hour Counter"),
                min = Counter("Minute Counter"), 
                sec= Counter("Second Counter")):
        self.hr = hr
        self.min = min
        self.sec = sec

        
    def Restart(self):
        self.hr.Reset()
        self.min.Reset()
        self.sec.Reset()


    def Tick(self):
        self.sec.Increment()

        if (self.sec.Ticks() == 60):
            self.sec.Reset()
            self.min.Increment()
            
            if (self.min.Ticks() == 60):
            
                self.min.Reset()
                if (self.hr.Ticks() < 24):
                
                    self.hr.Increment()
                
                else:
                
                    self.hr.Reset()
                
    def Display(self):
        h = f"{self.hr.Ticks():02}"
        m = f"{self.min.Ticks():02}"
        s = f"{self.sec.Ticks():02}"

        return f"{h}:{m}:{s}"

            

        
    

MyClock = Clock()

tracemalloc.start()
start_time = time.perf_counter()


for i in range(10000):
    MyClock.Tick()

end_time = time.perf_counter()



print (MyClock.Display())
MyClock.Restart()
print (MyClock.Display())


print(f"Elapsed Time: {(end_time - start_time) * 1000:.2f} ms")

current, peak = tracemalloc.get_traced_memory()
print(f"Current memory usage: {current / 1024:.2f} KB")
print(f"Peak memory usage: {peak / 1024:.2f} KB")
tracemalloc.stop()

File: Submissions/Submission12/Task_12/ClockC#/Clock.cs
using System.Security.Cryptography.X509Certificates;

public class Clock
{
    private Counter HourCounter;
    private Counter MinuteCounter;
    private Counter SecondCounter;


    public Clock()
    {
        HourCounter = new Counter("Hour Counter");
        MinuteCounter = new Counter("Minute Counter");
        SecondCounter = new Counter("Seconds Counter");
    }


    public void Restart()
    {
        HourCounter.Reset();
        MinuteCounter.Reset();
        SecondCounter.Reset();

    }

    public void Tick()
    {
        SecondCounter.Increment();

        if (SecondCounter.Ticks == 60)
        {
            MinuteCounter.Increment();
            SecondCounter.Reset();
            if (MinuteCounter.Ticks == 60)
            {
                MinuteCounter.Reset();
                if (HourCounter.Ticks < 24)
                {
                    HourCounter.Increment();
                }
                else
                {
                    HourCounter.Reset();
                }

            }

        }
    }
    public string Display()
    {
        string h = HourCounter.Ticks.ToString("D2");
        string m = MinuteCounter.Ticks.ToString("D2");
        string s = SecondCounter.Ticks.ToString("D2");
        string time = h + ":" + m + ":" + s;

        return time;
    }
}




